Add first version of first set of MeSH mappings

rays22 opened the pull request:
This PR intends to add the first set of OBO Ontology (CL, UBERON, GO, OBA) term mappings to a set of MeSH terms.

rays22 committed:
Add first version of first set of MeSH mappings

This PR intends to add the first set of OBO Ontology (CL, UBERON, GO, OBA) term mappings to a set of MeSH terms.

jamesamcl commented:
Thanks @rays22 -

My initial comments:

* We should be using `skos:exactMatch` as with other sssom mapping sets rather than `owl:equivalentClass`
* Labels are missing from the tsv which makes this difficult to check
* Yaml header is missing from the tsv
* Other metadata also missing; e.g. were these all manual curation?
* Related to above, where do the confidence scores come from?

jamesamcl commented:
@matentzn would welcome your feedback too

rays22 commented:
> * We should be using `skos:exactMatch` as with other sssom mapping sets rather than `owl:equivalentClass`
> * Labels are missing from the tsv which makes this difficult to check
> * Other metadata also missing; e.g. were these all manual curation?
> * Related to above, where do the confidence scores come from?

@jamesamcl ,
The `mp_hp_example*` files that use `owl:equivalentClass` had been automatically created by the mapping-commons github  template when I created the repo. I will remove them.

I will investigate this:

> * Yaml header is missing from the tsv

rays22 committed:
Remove, add and update files

rays22 commented:
- [ ] When you create mapping registries, please make sure they are registered here: https://github.com/mapping-commons/mapping-commons.github.io/blob/main/mapping-server.yml

rays22 committed:
Add more mappings

rays22 committed:
Update mappings

This commit intends to
1. update some of the mappings, and
2. add new MeSH Concept term mappings.

rays22 committed:
Update mapping registry configuration file

matentzn commented on mappings/obo-mesh.sssom.yml:
```diff
@@ -0,0 +1,17 @@
+license: https://creativecommons.org/publicdomain/zero/1.0/
+mapping_provider: https://www.ebi.ac.uk
+mapping_set_group: obo_mesh_mappings
+mapping_set_id: obo_mesh
```
https://github.com/mapping-commons/mesh-mappings/issues/2

cthoyt commented:
FYI we already did a lot of mesh to OBO mappings in Biomappings (https://github.com/biopragmatics/biomappings) - it would be a shame not to reuse those

jamesamcl commented:
> FYI we already did a lot of mesh to OBO mappings in Biomappings (https://github.com/biopragmatics/biomappings) - it would be a shame not to reuse those

Agreed! Should we mirror them here as SSSOM?

cthoyt commented:
Why mirror them? They have a PURL (https://w3id.org/biopragmatics/biomappings/sssom/biomappings.sssom.tsv) for anyone who wants to use them. I think it's a bit misguided to use GitHub as a place for mirroring data (is that what's going on with all of the repos in this org? I disagree with this)

cthoyt commented on mappings/obo-mesh_embedded.sssom.tsv:
```diff
@@ -0,0 +1,1133 @@
+#subject_source: http://purl.obolibrary.org/obo/oba.owl
+subject_id	subject_label	predicate_id	object_id	object_label	mapping_justification	subject_source	object_source	confidence	mapping_date	author_id	object_uri
+CL:0000017	spermatocyte	skos:exactMatch	M0020280	Spermatocytes	semapv:ManualMappingCuration	CL	MeSH	1	2025-03-26	orcid:0000-0001-8314-2140	http://id.nlm.nih.gov/mesh/M0020280
+CL:0000018	spermatid	skos:exactMatch	M0020276	Spermatids	semapv:ManualMappingCuration	CL	MeSH	1	2025-03-26	orcid:0000-0001-8314-2140	http://id.nlm.nih.gov/mesh/M0020276
```
the mapping_date seems fake, these likely weren't all manually mapped on this date.

rays22 replied:
Yes, my friend, these were all manually mapped on 2025-03-26.

cthoyt replied:
wow! I am more than impressed. do you have a time estimate on how long this took?

rays22 replied:
@cthoyt ,

Thank you for your appreciation of my work.

Here is my curation process:
1. I collected the textual definitions of the targeted list of MeSH Concepts, e.g. https://id.nlm.nih.gov/mesh/M0001044.html
2. I created several draft iterations of the term mappings using [OLS](https://www.ebi.ac.uk/ols4/ontologies/uberon/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FUBERON_0001876) and the MeSH Concept labels together with their definitions.
3. I double-checked and finalised the mappings for a set of 1,086 terms on 2025-03-26, believed to be correct at that time.

Step 2 took several days to weeks. Nevertheless, the final version of the manual mappings was created on 2025-03-26 (hence the value of `mapping_date = 2025-03-26`).
